### DISCUSSION TITLE
Support strict mode in PowerShell helpers?

### DISCUSSION THREAD

**Discussion by TheCakeIsNaOH** [chocolatey/choco#2483] [General]
I've noticed a couple of packages that use `Set-StrictMode`, but disable it when using choco's built in helpers, like `Get-ChocolateyWebFile`
Example: https://community.chocolatey.org/packages/AzurePowerShell#files

I've not seen any information about why it is disabled, or what specific issues have been found with the Chocolatey helpers. 

Is supporting `Set-StrictMode` something that would be feasible? I'm not sure what it would require.

**Comment by gep13** [comment]
In all honesty, I am not sure 😢 

@vexx32 @corbob @JPRuskin @Windos @pauby  can any of you provide input here?  Thanks

**Comment by pauby** [comment]
Just for reference here is the [Set-StrictMode page for Windows PowerShell 5.1](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/set-strictmode?view=powershell-5.1)

I'll pick this up internally and come back.

**Reply by pauby** [reply]
This is a much bigger effort than just adding it and seeing if it works (not that I'm suggesting that's what was being asked). After speaking earlier to the team, an internal issue has been raised to track the work.
